src/packages/src/xroad/common/base/usr/share/xroad/scripts/_backup_xroad.sh

```diff
@@ -35,21 +35,21 @@ create_database_backup () {
 }
 
 create_backup_tarball () {
+
   if [ "$ENCRYPT_MODE" = "encrypt" ] || [ "$ENCRYPT_MODE" = "signonly" ] ; then
+    local ENCRYPTION_ARGS=()
     if [ "$ENCRYPT_MODE" = "encrypt" ] ; then
       echo "CREATING ENCRYPTED AND SIGNED TAR ARCHIVE TO ${BACKUP_FILENAME}"
       local PUBCOUNT=0
-      if [ -d "${PUBKEYS_FOLDER}" ] ; then
-        local ENCRYPTION_ARGS
-
-        if [ -d "$DIR" ] ; then
-          for keyfile in "$PUBKEYS_FOLDER"/* ; do
-            (( PUBCOUNT++ )) || true
-            ENCRYPTION_ARGS+=(-f "${keyfile}")
-          done
-        fi
-        echo "Encrypting archive with servers public key and $PUBCOUNT extra public keys"
-        local SELF_ARGS=(--encrypt -r "${SECURITY_SERVER_ID}")
+      ENCRYPTION_ARGS=(--encrypt --trust-model direct --cipher-algo AES256 --no-auto-key-locate -r "${SECURITY_SERVER_ID}")
+      if [ -n "${GPG_KEYIDS}" ] ; then
+        local recipients=()
+        IFS=", " read -ra recipients <<< "${GPG_KEYIDS}"
+        for keyid in "${recipients[@]}"; do
+          (( PUBCOUNT++ )) || true
+          ENCRYPTION_ARGS+=(-r "$keyid")
+        done
+        echo "Encrypting archive with servers public key and $PUBCOUNT extra recipients"
```

Comment:
Extra recipients are not required, so I don't see the point for a warning that is not even normally visible since the UI does not display the backup output.